src/MQTTClient.c

```diff
@@ -2079,6 +2079,11 @@ MQTTResponse MQTTClient_subscribeMany5(MQTTClient handle, int count, char* const
 			}
 			else
 			{
+				if (count < sub->qoss->count)
+				{
+					rc = MQTTCLIENT_FAILURE;
+					goto exit;
+				}
```

Comment:
This code is inserted before the declaration of the "current" variable in the block. Even though this works on newer compilers, the declaration should come first.